### PR TITLE
CppCheck: Removing unusedVariable warnings

### PR DESCRIPTION
**Pull request by kushalsingh007**
- Problem was due to VDEBUG being used and not having associated meaning.
- Replaced VDEBUG with DEBUG in places where warning was being shown.
- Part of fix for #480

### CODE REVIEW

**Comment by kushalsingh007** [comment]
Is this fine or should I suppress the warnings instead?

**Comment by Lotterleben** [comment]
Hi,
I want to make sure that I understand you correctly... What do you mean by "not having associated meaning?"
(I built the VDEBUG macro to slim down the massive amount of debug output that I needed sometimes, but not always. This is a messy fix so I'm all for improving it, bit I want to make sure that I understand what your plan is... )

**Comment by kushalsingh007** [comment]
Hi :)
What I meant by that was the following , currently in this file if you consider the #define statement for it at https://github.com/kushalsingh007/RIOT/blob/unused/sys/net/routing/aodvv2/reader.c#L30
if VERBOSE_DEBUG is 1 then AODV_DEBUG(__VA_ARGS__) is used ,but when VERBOSE_DEBUG is 0 then it does not have any associated thing to it , and so it causes unused variable warning.
So to remove the warning I changed it to DEBUG instead . If you think that I am mistaken here , I can suppress the warning instead.

**Comment by Lotterleben** [comment]
Ah, thank for the clarification! The problem with just using DEBUG instead is that this produces unnecessary debug output that doesn't really have a context. In my opinion, suppressing the warning makes more sense here. :)

**Comment by Lotterleben** [comment]
(Oh, but keep in mind that each suppressed warning has to be alongside a short comment that explains why it is suppressed...)

**Comment by kushalsingh007** [comment]
@Lotterleben : I have suppressed it and added explanation for it ( + done rebasing ) , kindly review the changes.

**Comment by kushalsingh007** [comment]
Kindly restart the build , it got stalled :(

**Comment by cgundogan** [comment]
kicked travis

**Comment by kushalsingh007** [comment]
```All is well :) ```

File: sys/net/routing/aodvv2/reader.c [hunk 1]
```diff
@@ -616,6 +618,7 @@ static enum rfc5444_result _cb_rerr_blocktlv_addresstlvs_okay(struct rfc5444_rea
 
     VDEBUG("%s()\n", __func__);
     VDEBUG("\tmessage type: %d\n", cont->type);
+    /* cppcheck-suppress Unused variable as VDEBUG is needed for better debugging output. */
```
Comment (Lotterleben) [inline sys/net/routing/aodvv2/reader.c]: One small nitpick: As far as I checked, this suppression is actually not necessary, since this is the place where nbuf is actually used, *if* it is used at all. :)

**Comment by Lotterleben** [comment]
Could you remove the last suppression (the one I commented on)? Otherwise, I'd say this PR is good to go.

File: sys/net/routing/aodvv2/reader.c (hunk 1 above)
Comment (kushalsingh007) [inline sys/net/routing/aodvv2/reader.c]: oops!  forgot to undo the changes while editing , sending the updated changes in a minute.

**Comment by kushalsingh007** [comment]
Done , and rebased after taking your comment into account.

**Comment by Lotterleben** [comment]
Perfect! ACK and go. Thank you for submitting this PR! :) :sparkles:

**Comment by OlegHahm** [comment]
Don't you want to hit the merge button, too?

**Comment by kushalsingh007** [comment]
Kindly restart the build please :)

**Comment by Lotterleben** [comment]
Kicked Travis again.. Not sure what its problem is right now :/ 
@OlegHahm who are yo referring to with “you”? ;)
